pkg/kn/commands/revision/revision_list.go

```diff
@@ -66,3 +86,26 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
 	revisionListFlags.AddFlags(revisionListCommand)
 	return revisionListCommand
 }
+
+func getRevisionInfo(args []string, namespace string, client serving.ServingV1alpha1Interface) (*v1alpha1.RevisionList, error) {
+
+	switch len(args) {
+	case 0:
+		revision, err := client.Revisions(namespace).List(v1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		return revision, nil
+
+	case 1:
+		revision, err := client.Revisions(namespace).List(v1.ListOptions{FieldSelector: "metadata.name=" + args[0]})
```

Comment:
is there any escaping that needs to be done for the name? i think there was some packages in apimachinery around generating field and label selectors.

Reply:
```
  var listOptions v1.ListOptions  
  switch len(args) {
  case 0:
    listOptions = v1.ListOptions{}
  case 1:
    listOptions.FieldSelector = fields.Set(map[string]string{"metadata.name": args[0]}).String()
  default:
    return nil, fmt.Errorf("'revision list' only accepts no arguments or a single revision name, not %d arguments as given", len(args))
  }
  revision, err := client.Revisions(namespace).List(listOptions)
  if err != nil {
    return nil, err
  }
  return revision, nil
```
@savitaashture

Reply:
Updated